### PR TITLE
Add teleport command

### PR DESCRIPTION
**Pull request by xsebek**
- add command `teleport :: robot -> (int * int) -> cmd ()` requiring god capability

In challenges, it is useful to be able to check the state of some remote position.
This gets you there.

### CODE REVIEW

File: src/Swarm/Language/Syntax.hs [hunk 1]
```diff
@@ -489,6 +492,7 @@ constInfo c = case c of
   Format -> functionLow 1
   Concat -> binaryOp "++" 6 R
   AppF -> binaryOp "$" 0 R
+  Teleport -> commandLow 2
```
Comment (byorgey) [inline src/Swarm/Language/Syntax.hs:495]: Shouldn't this be `commandLow 1`?  It only takes one argument.

**Review by byorgey — CHANGES_REQUESTED** [review]
This all looks good to me.  However, I might suggest creating a new `teleport` capability and using that instead of `god`.  The reason is that I can imagine it would be fun to make a (very difficult to construct) teleportation device which actually provides this capability in the game.  We could of course use `god` for now and just add a new capability later, but using its own capability now will serve as a reminder that we want to eventually create a device for it.

File: src/Swarm/Language/Syntax.hs (hunk 1 above)
Comment (xsebek) [reply src/Swarm/Language/Syntax.hs:495]: It should, I am surprised that it worked. :open_mouth:

**Comment by xsebek** [comment]
OK, let's commit to adding a teleportation device :smile:

As a side note, I am hoping that `halt` (#392) would allow us to stop base and reprogram it to teleport, so in-world portals operated by system robots become possible :crossed_fingers:

**Comment by byorgey** [comment]
Ah, I see, yes, having `teleport : robot -> int * int -> cmd ()` does seem strictly more useful/interesting.

**Review by byorgey — APPROVED** [review]

File: src/Swarm/Game/Step.hs [hunk 2]
```diff
@@ -1346,7 +1346,7 @@ execConst c vs s k = do
       [VBool b] -> return $ Out (VBool (not b)) s k
       _ -> badConst
     Neg -> case vs of
-      [VInt n] -> return $ Out (VInt (-n)) s k
+      [VInt n] -> return $ Out (VInt (- n)) s k
```
Comment (xsebek) [inline src/Swarm/Game/Step.hs:1349]: OMG, I really have to find out which version of Fourmolu Restyled uses :expressionless:

**Review by byorgey — APPROVED** [review]

**Pull request merged**